### ISSUE TITLE
Should the interop supports 5.x PHP?

### ISSUE THREAD

**Issue by makasim** [queue-interop/queue-interop#2]
That's more a poll rather than an issue. So if you think it has to require PHP7.1 as put 👍 if you think it worth supporting 5.x put 👎 .

**Comment by pinepain** [comment]
I'd like to clarify that PHP 5.6 is EOL and it receives on security updates - http://php.net/supported-versions.php. I greatly respect those who various reasons can't just simply upgrade their codebase to 7.x, so the question is not to ignore that part of community, but clarify on roadmap: IMO, at the point when queue-interop become a by-law/de-facto standard, even 7.0 will be EOLed, so the question is about future goals rather then today.

**Comment by makasim** [comment]
Symfony 3.0 [requires 5.6](https://github.com/symfony/symfony/blob/3.0/composer.json#L19). 
Laravel 5.0 [requires 5.6](https://github.com/laravel/framework/blob/5.4/composer.json#L18) too 

If we migrate we will not be able to use the interop in any of those frameworks which is bad.

**Comment by pinepain** [comment]
To give a full perspective,

Symfony 4.0 [requires PHP 7.1](https://github.com/symfony/symfony/blob/e6f61504d70fe8a6bf50765c71869660d70ae9d5/composer.json#L19)
Laravel 5.5 [requires 7.0](https://github.com/laravel/framework/blob/5f71c09c2a079e865ec8420e2abe52633d57d310/composer.json#L18)

both coming later this year.

AFAIK, Laravel 5.0 is pretty mature and current stable is 5.4. It also have own Queue component which is just nice. Symfony 2.8 (which is current LTS) is also have plenty of queue bundles.

Giving that queue-interop is in it early development/discussion phase and doesn't have enough feedback from community, I would not consider that any mature and production-ready queue library will adopt it at current stage, neither new queue library that implement this interop pop up and gain enough eyes and adopters to be considered as a mature candidate for production use in such projects like Symfony, Laravel, Magento  with massive userbase. 

To sum up, early adopters naturally are more likely to use more recent PHP (7.x) version than those who will come when interface will be fully ready and will have enough implementations.

**Comment by makasim** [comment]
We already use it in several projects. All projecst are run on PHP 7 though not all use recent Symfony version. The oldest one is on 2.8.

**Comment by pinepain** [comment]
No, it's pretty valid point. While older `queue-interop` are still support 5.x it should not be a problem as current users may still use latest stable version that has 5.x support. And the overall question is about future, less likely there is a good reason to drop 5.x support in near future - you can still have a  feature branch which will adopt 7.x and port 5.x changes to it.

**Comment by makasim** [comment]
I am afraid it will be way over my head to support two branches at the same time. If some one wants to maintain the 7.x branch I dont mind.

**Comment by pinepain** [comment]
I gave a brief look on existent PSR's interfaces and it seems that vast majority of them are really support PHP 5.x (some of them even 5.3). Taking into account widespread of PHP 5.x, if interop want to be real interop it will have to deal with 5.x. I'd personally would love to see new cool features, but maybe some other time. Anyway, let's try to hear from others and keep this discussion live.

**Comment by makasim** [comment]
Conclusion: We should strive to support as a wider range of PHP version as possible.

**Comment by makasim** [comment]
As mentioned in #9 

> PHP 5 vs PHP 7:

I saw you discussed the issue here (#2). You decided to go for PHP 5 support a few month ago, but the tide has turned since then. Then newly passed PSR-15 is PHP 7+ only and upcoming PSR will be PHP 7+. So you should definitely target PHP 7 now.


So it make sense to drop php5 support
